app/models/subscriber_list.rb

```diff
@@ -16,6 +16,12 @@ class SubscriberList < ApplicationRecord
   has_many :matched_content_changes
   has_many :matched_messages
 
+  before_validation do
+    tags_or_links = links.any? ? links : tags
+    data_type = links.any? ? :links : :tags
+    self.tags_links_digest = HashDigest.generate(tags_or_links, data_type)
```

Comment:
As there isn't validation that ensures you can only have links or tags I wouldn't suggest that we have this based on or conditions.

I think it'd be make more sense to hash both links and tags as columns in the database, which seems to be a less domain specific approach to the optimisation.